src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -0,0 +1,693 @@
+  ∃y∈s, edist x y < r :=
+let ⟨t, ⟨ht, tr⟩⟩ :=  Inf_lt_iff.1 h in
+let ⟨y, ⟨ys, hy⟩⟩ := (mem_image _ _ _).1 ht in
+⟨y, ys, by rwa ← hy at tr⟩
+
+/-- The edist of `x` to `s` is bounded by the sum of the edist of `y` to `s` and
+the edist from `x` to `y` -/
+lemma inf_edist_le_inf_edist_add_edist : inf_edist x s ≤ inf_edist y s + edist x y :=
+begin
+  have : ∀z ∈ s, Inf (edist x '' s) ≤ edist y z + edist x y := λz hz, calc
+    Inf (edist x '' s) ≤ edist x z :
+      Inf_le ((mem_image _ _ _).2 ⟨z, hz, by refl⟩)
+    ... ≤ edist x y + edist y z : edist_triangle _ _ _
+    ... = edist y z + edist x y : add_comm _ _,
+  have : (λz, z + edist x y) (Inf (edist y '' s)) = Inf ((λz, z + edist x y) '' (edist y '' s)),
+  { refine Inf_of_continuous _ _ (by simp),
+    { exact continuous_add continuous_id continuous_const },
+    { assume a b h, simp, apply add_le_add_right' h }},
+  simp only [inf_edist] at this,
+  rw [inf_edist, inf_edist, this, ← image_comp],
+  simpa only [and_imp, function.comp_app, lattice.le_Inf_iff, exists_imp_distrib, ball_image_iff]
+end
+
+/-- The edist to a set depends continuously on the point -/
+lemma continuous_inf_edist : continuous (λx, inf_edist x s) :=
+continuous_of_le_add_edist 1 (by simp) $
+  by simp only [one_mul, inf_edist_le_inf_edist_add_edist, forall_2_true_iff]
+
+/-- The edist to a set and to its closure coincide -/
+lemma inf_edist_closure : inf_edist x (closure s) = inf_edist x s :=
+begin
+  refine le_antisymm (inf_edist_le_inf_edist_of_subset subset_closure) _,
+  refine ennreal.le_of_forall_epsilon_le (λε εpos h, _),
+  have εpos' : (0 : ennreal) < ε := by simpa,
+  have : inf_edist x (closure s) < inf_edist x (closure s) + ε/2 :=
+    ennreal.lt_add_right h (ennreal.half_pos εpos'),
+  rcases exists_edist_lt_of_inf_edist_lt this with ⟨y, ycs, hy⟩,
+  -- y : α,  ycs : y ∈ closure s,  hy : edist x y < inf_edist x (closure s) + ↑ε / 2
+  rcases emetric.mem_closure_iff'.1 ycs (ε/2) (ennreal.half_pos εpos') with ⟨z, zs, dyz⟩,
+  -- z : α,  zs : z ∈ s,  dyz : edist y z < ↑ε / 2
+  calc inf_edist x s ≤ edist x z : inf_edist_le_edist_of_mem zs
+        ... ≤ edist x y + edist y z : edist_triangle _ _ _
+        ... ≤ (inf_edist x (closure s) + ε / 2) + (ε/2) : add_le_add' (le_of_lt hy) (le_of_lt dyz)
+        ... = inf_edist x (closure s) + ↑ε : by simp [ennreal.add_halves]
+end
+
+/-- A point belongs to the closure of `s` iff its infimum edistance to this set vanishes -/
+lemma mem_closure_iff_inf_edist_zero : x ∈ closure s ↔ inf_edist x s = 0 :=
+⟨λh, by rw ← inf_edist_closure; exact inf_edist_zero_of_mem h,
+λh, emetric.mem_closure_iff'.2 $ λε εpos, exists_edist_lt_of_inf_edist_lt (by rwa h)⟩
+
+/-- Given a closed set `s`, a point belongs to `s` iff its infimum edistance to this set vanishes -/
+lemma mem_iff_ind_edist_zero_of_closed (h : is_closed s) : x ∈ s ↔ inf_edist x s = 0 :=
+begin
+  have := @mem_closure_iff_inf_edist_zero _ _ x s,
+  rwa closure_eq_iff_is_closed.2 h at this
```

Comment:
This is fine, but another style you might find useful is something like (untested)
```lean
begin
  convert mem_closure_iff_inf_edist_zero,
  exact closure_eq_iff_is_closed.2 h
end
```
If the direction of the goal after `convert` is wrong to apply `closure_eq_iff_is_closed`, you can reverse it using `convert ← mem_closure_iff_inf_edist_zero`.

Reply:
Nice, thanks! Done.